src/mongo_client.ts

```diff
@@ -475,66 +519,53 @@ export class MongoClient extends TypedEventEmitter<MongoClientEvents> {
 
     const force = typeof forceOrCallback === 'boolean' ? forceOrCallback : false;
 
-    return maybePromise(callback, callback => {
+    return maybeCallback(async () => {
       if (this.topology == null) {
         // Do not connect just to end sessions
-        return callback();
+        return;
       }
 
       const activeSessionEnds = Array.from(this.s.activeSessions, session => session.endSession());
       this.s.activeSessions.clear();
 
-      Promise.all(activeSessionEnds)
-        .then(() => {
-          if (this.topology == null) {
-            return;
-          }
-          // If we would attempt to select a server and get nothing back we short circuit
-          // to avoid the server selection timeout.
-          const selector = readPreferenceServerSelector(ReadPreference.primaryPreferred);
-          const topologyDescription = this.topology.description;
-          const serverDescriptions = Array.from(topologyDescription.servers.values());
-          const servers = selector(topologyDescription, serverDescriptions);
-          if (servers.length === 0) {
-            return;
-          }
-
-          const endSessions = Array.from(this.s.sessionPool.sessions, ({ id }) => id);
-          if (endSessions.length === 0) return;
-          return this.db('admin')
+      await Promise.all(activeSessionEnds);
+
```

Comment:
Are we missing a check for a null topology here?  There used to be one

Reply:
We should be able to move the one above to below here, the only way it would become null is if there was parallel closes I guess?